server/src/instant/db/model/attr_pat.clj

```diff
@@ -306,14 +309,16 @@
       (throw-invalid-data-value! state attr attr-data-type value)
       value)))
 
-(defn validate-value-type! [state attr data-type v]
-  (doseq [v (if (set? v) v [v])
-          :let [v (if (and (map? v)
-                           (contains? v :$not))
-                    (:$not v)
-                    v)]]
-    (throw-on-invalid-value-data-value! state attr data-type v))
-  v)
+(defn coerce-v-single! [state attr data-type v]
+  (if (and (map? v)
+           (contains? v :$not))
+    (update v :$not coerce-value-data-value! state attr data-type)
```

Comment:
I think you might need to do something like `(update v :$not (partial coerce-value-data-value! state attr data-type))` to get the arg in the right position.